rcongui/src/components/Header/serverList.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import InputLabel from "@material-ui/core/InputLabel";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import Grid from "@material-ui/core/Grid";
+import { showResponse } from "../../utils/fetchUtils";
+
+
+const Selector = ({
+    classes,
+    defaultValue,
+    defaultText,
+    values,
+    currentValue,
+    onChange,
+    kind
+  }) => (
+  <FormControl className={classes.logsControl}>
+    <InputLabel shrink>{kind}</InputLabel>
+    <Select
+      value={currentValue}
+      onChange={e => onChange(e.target.value)}
+      displayEmpty
+    >
+      {defaultValue !== undefined ? (
+        <MenuItem value={defaultValue}>
+          <em>{defaultText}</em>
+        </MenuItem>
+      ) : (
+          ""
+        )}
+      {values.map(a => (
+        <MenuItem value={a.port}>{a.title}</MenuItem>
```

Comment:
add a `key={a.port}` property, otherwise react is not happy

Reply:
key= and value= ?

Reply:
It's just to give the html node a unique key so that react knows if it should re-render it the next time. It's an optimization thing